modules/basegl/src/processors/imageprocessing/imagechannelcombine.cpp

```diff
@@ -0,0 +1,120 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2013-2019 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/basegl/processors/imageprocessing/imagechannelcombine.h>
+#include <modules/opengl/texture/textureunit.h>
+#include <modules/opengl/texture/textureutils.h>
+#include <modules/opengl/shader/shaderutils.h>
+
+namespace inviwo {
+
+    const ProcessorInfo ImageChannelCombine::processorInfo_{
+        "org.inviwo.ImageChannelCombine",  // Class identifier
+        "Image Channel Combine",           // Display name
+        "Image Operation",                 // Category
+        CodeState::Experimental,           // Code state
+        Tags::GL,                          // Tags
+    };
+    const ProcessorInfo ImageChannelCombine::getProcessorInfo() const { return processorInfo_; }
+
+    ImageChannelCombine::ImageChannelCombine() :
+        Processor()
+        , inport0_("inport0", true)
+        , inport1_("inport1", true)
+        , inport2_("inport2", true)
+        , inport3_("inport3", true)
+        , outport_("outport", false)
+        , alpha_("alpha", "Alpha", 1.0f, 0.0f, 1.0f, 0.001f)
+        , shader_("img_channel_combine.frag")
```

Comment:
this will initialize the shader of the base class! Use appropriate constructor instead.